digdag-standards/src/main/java/io/digdag/standards/operator/state/PollingWaiter.java

```diff
@@ -87,9 +123,20 @@ public PollingWaiter withPollInterval(DurationInterval retryInterval)
     public <T> T await(Operation<Optional<T>> f)
     {
         TaskState pollState = state.nestedState(stateKey);
-
+        Long startTime = pollState.params().get(START_TIME, Long.class, null);
+        if (startTime == null) {
+            startTime = System.currentTimeMillis();
+            pollState.params().set(START_TIME, startTime);
+        }
         TaskState operationState = pollState.nestedState(OPERATION);
 
+        //Check timeout
+        Long now = System.currentTimeMillis();
+        if (timeout.isPresent() && timeout.get().toMillis() <= now - startTime) {
+            logger.debug("Timeout happened. startTime:{}, timeout:{}", startTime, timeout.get());
```

Comment:
Please reconsider whether this log is for users or you only, while s3_wait operator itself has its own logging. If you only, then use `trace` or remove.